recipes/patchelf/all/conanfile.py

```diff
@@ -0,0 +1,54 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.33.0"
+
+class PatchElfConan(ConanFile):
+    name = "patchelf"
+    description = "A small utility to modify the dynamic linker and RPATH of ELF executables"
+    topics = ("conan", "elf", "linker", "interpreter", "RPATH", "binaries")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/NixOS/patchelf"
+    license = "GPL-3.0-or-later"
+    settings = "os", "arch", "compiler", "build_type"
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def build_requirements(self):
+        self.build_requires("libtool/2.4.6")
+
+    def configure(self):
+        if not tools.is_apple_os(self.settings.os) and self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("PatchELF is only available for GNU-like operating systems (e.g. Linux)")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self)
+        self._autotools.configure(configure_dir=self._source_subfolder)
+        return self._autotools
+
+    def build(self):
+        with tools.chdir(self._source_subfolder):
+            self.run("{} -fiv --warnings=all".format(tools.get_env("AUTORECONF")), run_environment=True)
```

Comment:
I've added `--warnings=all` from patchelf's [build script](https://github.com/NixOS/patchelf/blob/master/bootstrap.sh) to be consistent with patchelf's build process. This flag is not strictly required, though. We can discuss removing it.

Reply:
As long as there's no `--warnings=error` :smile:

Reply:
@madebr Random question but I noticed the usage of autoreconf to be really varied...
https://github.com/conan-io/conan-center-index/search?p=3&q=AUTORECONF

Things like https://github.com/conan-io/conan-center-index/blob/41d4fbb8c218f9d938ccb8b9d888bff52002e62e/recipes/cunit/all/conanfile.py#L112 just look wrong and https://github.com/conan-io/conan-center-index/blob/d28a79b253a674e86e7296c7daef89336bd4f96a/recipes/libsodium/all/conanfile.py#L95
doesn't match the reset

❓ 
- `tools.get_env` vs `os.environ`
- is `run_environment=True` required/helpful?
- is `win_bash=tools.os_info.is_windows` required for building on Windows?

➡️ I want to update our "how to add packages" docs with the copy gnuconfig stuff and this but I am not sure what a good example looks like

Reply:
Here is the man page of `autoreconf`: https://linux.die.net/man/1/autoreconf
`-f` is to "force" rewrite all files (mark all files as out of date) and `-v` is for verbose output.
So `autoreconf -i` should be enough. I've started using `-fiv` just to be sure. I've encountered these options also in random utility scripts of projects to re-create their configure scripts.

- We've started to converge around `tools.get_env("AUTORECONF")`. Other usages are from the "old times" where that utility function did not exist yet or from copy-paste.
A contra for using `tools.get_env("AUTORECONF")` is that when `autoconf` is not a build requirement when cross building, then it tries to run `None -fiv` instead.
- `run_environment=True` might be useful: autoconf is a combination of perl and m4 scripts. Both can run executables.
Or perl might be a conan package, needing a run environment. (I've started working on a perl recipe, but it's unfinished)
- `win_bash=tools.os_info.is_windows` is mostly needed to interpret the [shebang at the top of autoreconf](https://github.com/autotools-mirror/autoconf/blob/64df9b4523fe3b3c5d6e444fb65b1d92fc5073ce/bin/autoreconf.in#L1). Autoconf is a GNU project that might expect forward slashes and other unix tools present. I don't think a bash script is run. I'm not sure though. It runs fine on Windows with strawberryperl.